.github/workflows/coverage-check.yml

```diff
@@ -0,0 +1,79 @@
+name: Coverage Check
+
+on:
+  pull_request:
+    branches: [ main ]
+
+permissions:
+  contents: read
+  pull-requests: write
```

Comment:
There seems to be no sense in having `write` permissions here. Do you know why they are needed? I don't see how this workflow modifies anything about PRs.